Sources/ComposableArchitecture/Reducer/ReducerBuilder.swift

```diff
@@ -7,175 +7,101 @@
-    where R.State == State, R.Action == Action {
-      Reduce(reducer)
+    public static func buildArray(
+      _ reducers: [some ReducerProtocol<State, Action>]
+    ) -> some ReducerProtocol<State, Action> {
+      _SequenceMany(reducers: reducers)
     }
-  #endif
-
-  @inlinable
-  public static func buildLimitedAvailability<R: ReducerProtocol>(
-    _ wrapped: R
-  ) -> _Optional<R>
-  where R.State == State, R.Action == Action {
-    _Optional(wrapped: wrapped)
-  }
-
-  @inlinable
-  public static func buildOptional<R: ReducerProtocol>(_ wrapped: R?) -> _Optional<R>
-  where R.State == State, R.Action == Action {
-    _Optional(wrapped: wrapped)
-  }
-
-  @inlinable
-  public static func buildPartialBlock<R: ReducerProtocol>(first: R) -> R
-  where R.State == State, R.Action == Action {
-    first
-  }
-
-  @inlinable
-  public static func buildPartialBlock<R0: ReducerProtocol, R1: ReducerProtocol>(
-    accumulated: R0, next: R1
-  ) -> _Sequence<R0, R1>
-  where R0.State == State, R0.Action == Action {
-    _Sequence(accumulated, next)
-  }
-
-  public enum _Conditional<First: ReducerProtocol, Second: ReducerProtocol>: ReducerProtocol
-  where
-    First.State == Second.State,
-    First.Action == Second.Action
-  {
-    case first(First)
-    case second(Second)
 
     @inlinable
-    public func reduce(into state: inout First.State, action: First.Action) -> EffectTask<
-      First.Action
-    > {
-      switch self {
-      case let .first(first):
-        return first.reduce(into: &state, action: action)
-
-      case let .second(second):
-        return second.reduce(into: &state, action: action)
-      }
+    public static func buildBlock() -> some ReducerProtocol<State, Action> {
+      EmptyReducer()
     }
-  }
 
-  public struct _Optional<Wrapped: ReducerProtocol>: ReducerProtocol {
-    @usableFromInline
-    let wrapped: Wrapped?
+    @inlinable
+    public static func buildBlock(
+      _ reducer: some ReducerProtocol<State, Action>
+    ) -> some ReducerProtocol<State, Action> {
+      reducer
+    }
 
-    @usableFromInline
-    init(wrapped: Wrapped?) {
-      self.wrapped = wrapped
+    @inlinable
+    public static func buildEither<R0: ReducerProtocol, R1: ReducerProtocol>(
+      first reducer: R0
+    ) -> _Conditional<R0, R1>
+    where R0.State == State, R0.Action == Action, R1.State == State, R1.Action == Action {
+      .first(reducer)
     }
 
     @inlinable
-    public func reduce(
-      into state: inout Wrapped.State, action: Wrapped.Action
-    ) -> EffectTask<Wrapped.Action> {
-      switch wrapped {
-      case let .some(wrapped):
-        return wrapped.reduce(into: &state, action: action)
-      case .none:
-        return .none
-      }
+    public static func buildEither<R0: ReducerProtocol, R1: ReducerProtocol>(
+      second reducer: R1
+    ) -> _Conditional<R0, R1>
+    where R0.State == State, R0.Action == Action, R1.State == State, R1.Action == Action {
```

Comment:
`buildEither` was the one spot I couldn't get `some ReducerProtocol` to work...so there may be some inference issues when it comes to `if`-`else`...but that's rare and I guess we can revisit if/when it comes up.

Reply:
@stephencelis were you using `_Conditional<R0, R1>.first(reducer)`?

Reply:
I was trying to, but couldn't figure out how to get `R0` to be a `some` and `R1` to be a generic that propagates.

Reply:
Oh I get you. 
The only way I can think of at the moment would be to use custom inits. eg: ` _Conditional(first: reducer, secondType: R1.self)` but not sure if that's any better than just passing the two generics.

Reply:
Fun idea! Seems to work and I imagine is better on the type checker than the alternative, so pushed, thanks! :smile: